### ISSUE TITLE
Pragma header should be added only once

### ISSUE THREAD

**Issue by pash7ka** [NomicFoundation/truffle-flattener#46]
While repeating `pragma` header does not make code invalid, it can look much better if only one `pragma` will be inserted at the top of the file. It may be taken from the target contract only, ignoring pragma on it's dependencies.

**Comment by alcuadrado** [comment]
Hey @pash7ka, thanks for creating this issue!

This used to be the case, but unifying the pragma can get very tricky. For example: If you have multiple pragmas, which one should you use? Are they compatible?

**Comment by pash7ka** [comment]
Hey @alcuadrado,
I guess most uses of truffle-flattener are for source code verification purposes. In this case it's still required for the user to select one version of the compiler, used to compile contract he verifies.
I believe, for most cases it will be the first contract specified in arguments for truffle-flattener. So that's the _pragma solidity_ line i propose to use.
For other pragmas, like _pragma experimental_ i propose to move this line to the top of the flattened contract. This way it will be easy to remove/change them if necessary.

Also, it would be nice to support an option to specify target compiler version and generate _pragma solidity_ for it, instead of any specified in the source files. This may be useful for tools like [openzeppelin verify](https://docs.openzeppelin.com/cli/2.6/commands#verify) which use truffle-flattener under-hood.

**Comment by alcuadrado** [comment]
_Closing to clean up the repository. If this problem persists, please open a new issue with a clear explanation._
